scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/LazyTokenIterator.scala

```diff
@@ -118,7 +118,6 @@ private[parsers] class LazyTokenIterator private (
   def observeOutdented(): Boolean =
     dialect.allowSignificantIndentation && (curr.regions match {
       case (region: SepRegionIndented) :: tail if (curr.token match {
-            case _: KwThen | _: KwElse | _: KwDo | _: KwYield => true
```

Comment:
Och, interesting. I think the main parser in the compiler handles it this way :thinking:

Reply:
i am trying to get rid of these methods (because they are invoked by the parser) and make sure that the token iterator returns indent/outdent itself. this is just the first step. not sure if that makes it clearer.

Reply:
ah :) no strong opinion there. also, saw how hard it was with the fewer-braces change that martin proposed recently, took a while.